packages/veritone-react-common/src/components/ExpandableSearchField/index.js

```diff
@@ -0,0 +1,105 @@
+import React from 'react';
+import { func } from 'prop-types';
+
+import TextField from '@material-ui/core/TextField';
+import IconButton from '@material-ui/core/IconButton';
+import SearchIcon from '@material-ui/icons/Search';
+import CloseIcon from '@material-ui/icons/Close';
+
+import Grow from '@material-ui/core/Grow';
+
+import cx from 'classnames';
+import styles from './styles.scss';
+
+class ExpandableSearchField extends React.Component {
+  static propTypes = {
+    onSearch: func
+  };
+
+  state = {
+    expanded: false,
+    search: ''
+  };
+
+  resetSearch = evt => {
+    this.setState({
+      search: '',
+      expanded: true
+    });
+    if (this._inputRef) {
+      this._inputRef.focus();
+    }
+  };
+
+  showSearchBar = evt => {
+    this.setState({
+      expanded: true
+    });
+  };
+
+  hideSearchBar = evt => {
+    if (!this.state.search || this.state.search.trim().length === 0) {
+      this.setState({
+        expanded: false
+      });
+    }
+  };
+
+  onChange = evt => {
+    this.setState({
+      search: evt.target.value
+    });
+  };
+
+  onKeyPress = evt => {
+    if (
+      evt.key === 'Enter' &&
+      this.props.onSearch &&
+      evt.target.value &&
+      evt.target.value.trim().length > 0
+    ) {
+      this.props.onSearch(evt.target.value.trim());
```

Comment:
pretty sure trimming should be done by the _consumer_ of the search, since there may be a case where whitespace is important. More generic that way, and consistent with other html inputs.